src/cargo/core/workspace.rs

```diff
@@ -103,7 +103,7 @@ pub struct Workspace<'gctx> {
 
     /// The resolver behavior specified with the `resolver` field.
     resolve_behavior: ResolveBehavior,
-    resolve_honors_rust_version: bool,
+    resolve_honors_rust_version: Option<bool>,
```

Comment:
Wonder why we need an `Option` here. It doesn't seem too relevant to this PR.

Reply:
Oops, that represents a bug.  Fixing it.

Reply:
I misread, there wasn't a bug and I didn't need to take advantage of this.  Simplifying the code.